runtime/lua/vim/lsp/diagnostic.lua

```diff
@@ -110,14 +110,22 @@ local function diagnostic_lsp_to_vim(diagnostics, bufnr, client_id)
   return vim.tbl_map(function(diagnostic)
     local start = diagnostic.range.start
     local _end = diagnostic.range['end']
+    local message ---@type string
+    if type(diagnostic.message) == 'string' then
+      ---@cast diagnostic { message: string }
+      message = diagnostic.message
+    else
+      vim.lsp.log.info(string.format('Unsupported Markup message from LSP client %d', client_id))
```

Comment:
warn?

Reply:
I'd tend to turn this even into an error using `vim.notify_once`. If a server returns markdown diagnostic messages without the client advertising the capability it is clearly a bug in the server

Reply:
Sounds good. I used `log` because that's what we use for unknown `DiagnosticTag`s, which are technically really bad since the supported set is also announced via capabilities.